lib/iex/lib/iex/evaluator.ex

```diff
@@ -58,17 +58,37 @@ defmodule IEx.Evaluator do
   # https://github.com/elixir-lang/elixir/issues/1089 for discussion.
   @break_trigger "#iex:break\n"
 
+  @space " "
+  @leading_binary_operators ~w(|> * / ++ -- <= >= <> > < && || === !== == != =~ or and in)
```

Comment:
There are some operators missing, such as `<<<`, which is going to complicate your leading binary check. :) See elixir_parser.yrl for a complete list.

Reply:
```elixir
["not in" | ~w(-> <- when :: | => & = || ||| or && &&& and == != =~ === !==
 < > <= >= |> <<< >>> <<~ ~>> <~ ~> <~> <|> in ^^^ ++ -- .. <>
+++ --- + - * / + - ! ^ not ~~~ . @)] -- 
["not in", "or", "and", "in"] ++ ~w(
    ^^^ <<< >>> <<~ ~>> <~ ~> <~> <|> &&& ||| === !==
    |> ++ -- <= >= <> > < && || == != =~
    * /)
#⇒ ["->", "<-", "when", "::", "|", "=>", "&", "=", "..", "+++", "---",
#   "+", "-",
#   "+", "-", "!", "^", "not", "~~~", ".", "@"]
```